src/components/Card.js

```diff
@@ -1,34 +1,54 @@
 class Card {
-  constructor(data, cardSelector, handleCardClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    isLiked
+  ) {
     this._name = data.name;
     this._link = data.link;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDelete = handleDeleteClick;
+    this._handleLike = handleLikeClick;
+    this._likes = Array.isArray(data.likes) ? data.likes : [];
+    this._cardID = data.id;
+    this._isLiked = isLiked;
+  }
+
+  isLiked() {
+    // Return true if the user liked the card, otherwise false
+    return this._isLiked;
```

Comment:
You should calculate this value here using the user id `this._likes.some(like => like._id === this._userId)`